std/string.d

```diff
@@ -5804,24 +5804,25 @@ unittest
  * Returns:
  *     $(D bool)
  */
-
-bool isNumeric(const(char)[] s, in bool bAllowSep = false) @safe pure
+bool isNumeric(S)(S s, bool bAllowSep = false) if (isSomeString!S)
 {
-    import std.algorithm : among;
+    import std.algorithm.comparison : among;
 
-    immutable iLen = s.length;
-    if (iLen == 0)
+    if (s.empty)
         return false;
 
     // Check for NaN (Not a Number) and for Infinity
     if (s.among!((a, b) => icmp(a, b) == 0)
             ("nan", "nani", "nan+nani", "inf", "-inf"))
         return true;
 
-    immutable j = s[0].among!('-', '+')() != 0;
+    if (s[0] == '-' || s[0] == '+')
+        s.popFront;
+
+    immutable iLen = s.length;
     bool bDecimalPoint, bExponent, bComplex, sawDigits;
 
-    for (size_t i = j; i < iLen; i++)
+    for (size_t i = 0; i < iLen; i++)
```

Comment:
Tried that, but `i` is modified in the loop in order to skip characters, so I didn't see a way to use foreach